### PR TITLE
Add SIM114 rule

### PR DESCRIPTION
**Pull request by TomeHirata**
[Open in GitHub Codespaces](https://codespaces.new/TomeHirata/mlflow/pull/9922?quickstart=1)

### Related Issues/PRs
https://github.com/mlflow/mlflow/issues/9834

### What changes are proposed in this pull request?

This PR is for adding SIM114	if-with-same-arms Ruff Rule

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/search_utils.py [hunk 1]
```diff
@@ -869,11 +869,11 @@ class SearchExperimentsUtils(SearchUtils):
 
     @classmethod
     def _invalid_statement_token_search_experiments(cls, token):
-        if isinstance(token, Comparison):
-            return False
-        elif token.is_whitespace:
-            return False
-        elif token.match(ttype=TokenType.Keyword, values=["AND"]):
+        if (
```
Comment (TomeHirata) [inline mlflow/utils/search_utils.py:872]: comment: [this comment](https://github.com/mlflow/mlflow/pull/9919#discussion_r1356516598) is actually incorporated in this PR

File: mlflow/models/utils.py [hunk 2]
```diff
@@ -753,10 +753,9 @@ def _is_scalar(x):
     if not input_schema.is_tensor_spec():
         if isinstance(pf_input, (list, np.ndarray, dict, pd.Series, str, bytes)):
             try:
-                if isinstance(pf_input, (str, bytes)):
-                    pf_input = pd.DataFrame([pf_input])
-                elif isinstance(pf_input, dict) and all(
-                    _is_scalar(value) for value in pf_input.values()
+                if isinstance(pf_input, (str, bytes)) or (
+                    isinstance(pf_input, dict)
+                    and all(_is_scalar(value) for value in pf_input.values())
```
Comment (harupy) [inline mlflow/models/utils.py:758]:
```suggestion
                    and all(map(_is_scalar pf_input.values())
```

File: mlflow/types/utils.py [hunk 3]
```diff
@@ -345,13 +345,13 @@ def _validate_input_dictionary_contains_only_strings_and_lists_of_strings(data)
     for key, value in data.items():
         if not value_type:
             value_type = type(value)
-        if isinstance(key, bool):
+        if isinstance(key, bool) or not isinstance(key, (str, int)):
             invalid_keys.append(key)
-        elif not isinstance(key, (str, int)):
-            invalid_keys.append(key)
-        if isinstance(value, list) and not all(isinstance(item, (str, bytes)) for item in value):
```
Comment (harupy) [inline mlflow/types/utils.py:352]: `if` here should be `elif`. It's weird to append the same key twice to `invalid_keys`.

Comment (TomeHirata) [reply mlflow/types/utils.py:352]: Yes, I was confused with this but thought it was intentional. Let me fix it.

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**